File: cpp/Day11/attendancesystem/employee.cpp
#include "employee.h"

void loadData(vector<Employee *> &v1)
{
}

Employee::Employee()
{
    cout << "Employee()" << endl;
}
Employee::Employee(double salary)
{
    cout << "Employee(double)" << endl;
}
void Employee::accept()
{
    cout << "Employee::accept()" << endl;
}

void Employee::display()
{
    cout << "Employee::display()" << endl;
}

File: java/Day05/Demo03/src/com/sunbeam/p2/Employee.java
package com.sunbeam.p2;

import java.util.Scanner;

// Employee is-a Person
public class Employee extends Person {
	private int empid;
	private double salary;

	public Employee() {
	}

	public Employee(int empid, String name, double salary) {
		super(name); // parameterized ctor of super class is called
		this.empid = empid;
		this.salary = salary;
	}

	// Method Overriding
	public void accept() {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the empid -  ");
		empid = sc.nextInt();

		super.accept(); // person class accept()

		System.out.println("Enter the Salary -  ");
		salary = sc.nextDouble();
	}

	// Method Overriding
	public void display() {
		System.out.println("Empid - " + empid);

		super.display();

		System.out.println("Salary - " + salary);
	}

}
